core/tests/protractor_utils/AdminPage.js

```diff
@@ -272,6 +287,50 @@ var AdminPage = function() {
       expect(name).toEqual(expectedUsernamesArray[j]);
     }
   };
+
+  this.addReviewer = async function(name, newRole) {
+    await waitFor.elementToBeClickable(
+      adminRolesTab, 'Admin Roles tab is not clickable');
+    await adminRolesTab.click();
+
+    // Change values for "add reviewer for community" form, and submit it.
+    await waitFor.visibilityOf(
+      addReviewerName, 'Add reviewer name is not visible');
+    await addReviewerName.sendKeys(name);
+
+    var roleOption = selectReviewerRole.element(
+      by.cssContainingText('option', newRole));
+    await waitFor.visibilityOf(
+      roleOption, 'Reviewer role option is not visible');
+    await roleOption.click();
+
+    await addReviewerFormSubmit.click();
+    await waitFor.visibilityOf(
+      statusMessage, 'Confirmation message not visible');
+    await waitFor.textToBePresentInElement(
+      statusMessage, 'Successfully added ',
+      'Could not add reviewer successfully');
+  };
+
+  this.getReviewersAssignedToRole = async function(role) {
+    await waitFor.visibilityOf(reviewerRoleDropdown,
+      'Reviewer role dropdown taking too long to be visible');
+    await reviewerRoleDropdown.sendKeys('By Role');
+
+    await reviewerRoleValueOption.click();
+    await reviewerRoleValueOption.sendKeys(role);
+
+    await viewReviewerRoleButton.click();
+  };
+
+  this.expectReviewerUsernamesStrToMatch = async function(
+      expectedUsernamesArrayStr) {
+    // The usernames are displayed as a
+    // 'Usernames: ["username1", "username2"]' string.
+    var reviewerUsernamesArrayStr = await element(
+      by.css('.protractor-test-reviewer-roles-result')).getText();
```

Comment:
Can you make sure this is visible first or use the actions? Also, can you move the selector to the top of the file with the other ones?

Reply:
Ahh yes, I guess we don't have an action yet for `getText()`. You can just use the normal waitFor statements to check that it's visible then. If you like, you could also add a function to the actions file for `getText`

Reply:
Yeah I tried defining the reviewerUsernamesArrayStr up above with the rest of the elements and then first checking waitFor.visibilityOf of the reviewerUsernamesArrayStr and after that check is done, then getting the text using getText() but I kept getting the following error:
Error while waiting for Protractor to sync with the page: "both angularJS testability and angular testability are undefined.  This could be either because this is a non-angular page or because your test involves client-side navigation, which can interfere with Protractor's bootstrapping.  See http://git.io/v4gXM for details"
That's why I didn't change this part of the code. I'm emailing the e2e test team about it. What're your thoughts?

Reply:
@kaylahardie the `Error while waiting for Protractor to sync with the page: "both angularJS testability and angular testability are undefined.` flake generally occurs when the e2e tests try to perform angular operations on a non-angular page. This can happen for a few reasons:

1. An error prevents the page from loading
2. The test incorrectly ends up on the signup page, which is non-angular
3. A page isn't found (often because a topic or exploration ID is `null`)

To debug this, I would suggest trying to figure out what page is loaded when the test fails. If you can run it locally, you can just watch the test run or use the debugger. If it only occurs in the CI, you can enable screenshots and then SSH into the CircleCI box to get the screenshots (we can't SSH into TravisCI machines unfortunately)

Reply:
Thanks @U8NWXD! :) Sorry for the delayed response - do you have any tips on the tests failing right away due to the gravatar ssl certificate failing?

Reply:
@U8NWXD, I just commented out the gravatar code while I was testing it locally...I submitted my changes but sometimes the 'should run, verify and stop one-off jobs' test fails...I think it may be because of how that code uses get(i) so many times...that could be the result of the flakiness? I can look into that but in the meantime can you PTAL at my changes to see if my latest updates make sense? Thanks :)